packages/lib/src/Login.wc.svelte

```diff
@@ -1,26 +1,25 @@
 <svelte:options customElement="sesamy-login" />
 
 <script lang="ts">
+  import type { SesamyAPI } from "@sesamy/sesamy-js";
+  import Avatar from "./Avatar.wc.svelte";
   import Base from "./Base.svelte";
   import Button from "./Button.wc.svelte";
-  import type { Api } from "./api";
+  import type { LoginProps } from "./types";
 
-  let loading = false;
-  let loggedIn = false;
-  let userAvatar = null;
+  let { loading, loggedIn, userAvatar }: LoginProps = $props();
 
-  const login = async (api: Api) => {
+  const login = async (api: SesamyAPI) => {
     loading = true;
     try {
       await api.auth.loginWithRedirect();
     } catch (error) {
-      console.error("Login failed:", error);
-    } finally {
       loading = false;
+      console.error("Login failed:", error);
     }
   };
 
-  const logout = async (api: Api) => {
+  const logout = async (api: SesamyAPI) => {
```

Comment:
Use the SesamyAPI interface which is exported from sesamy-js in the last version :)